test/sys/test_signalfd.rs

```diff
@@ -0,0 +1,52 @@
+#[cfg(target_os = "linux")]
+#[test]
+// FIXME: Fails with "fatal runtime error: failed to initiate panic, error 5" on i686-unknown-linux-musl
+#[cfg_attr(all(target_env = "musl", target_arch = "x86"), ignore)]
+fn test_signalfd() {
+    use nix::sys::signalfd::{SFD_NONBLOCK, SignalFd};
+    use nix::sys::signal;
+    use nix::unistd;
+    use nix::unistd::*;
+    use nix::unistd::ForkResult::*;
+    use nix::sys::signal::*;
+    use nix::sys::wait::*;
+
+    // Set up a signalfd to listen for SIGUSR1 on this thread
+    let mut mask = SigSet::empty();
+    mask.add(SIGUSR1);
```

Comment:
`SIGUSR1` is a global resource, so this test will conflict with any other test that does anything with it.  I dealt with that in PR #638 by creating a mutex to protect `SIGUSR2`.  You should probably do the same for `SIGUSR1`